include/ndarray/pybind11.h

```diff
@@ -42,13 +42,131 @@
  */
 
 #include "pybind11.h"
+#include "pybind11/numpy.h"
 
-#include "Python.h"
 #include "ndarray.h"
-#include "ndarray/converter/numpy.h"
-#include "ndarray/converter/ufunctors.h"
-#include "ndarray/converter/Vector.h"
-#include "ndarray/converter/eigen.h"
+#include "ndarray/eigen.h"
+#include "ndarray/converter/PyManager.h"
+
+namespace ndarray {
+
+#if PYBIND11_VERSION_MAJOR == 2 && PYBIND11_VERSION_MINOR <= 1
+using pybind11_np_size_t = size_t;
+#else
+using pybind11_np_size_t = ssize_t;
+#endif
+
+template <typename T, int N, int C>
+struct Pybind11Helper {
+    using Element = typename ndarray::Array<T,N,C>::Element;
+    using Wrapper = pybind11::array_t<typename std::remove_const<Element>::type, 0>;  // 0: no ensurecopy
+    static constexpr bool isConst = std::is_const<Element>::value;
+
+    Pybind11Helper() : isNone(false), wrapper() {}
+
+    bool init(pybind11::handle src) {
+        isNone = src.is_none();
+        if (isNone) {
+            return true;
+        }
+        try {
+            wrapper = pybind11::reinterpret_borrow<Wrapper>(src);
+        } catch (pybind11::error_already_set & err) {
+            return false;
+        }
+        return true;
+    }
+
+    bool check() const {
+        if (!isNone) {
```

Comment:
Are you sure?

Reply:
Good catch, need to reverse the sign of this!